### PR TITLE
feat: Update Tracing Call Types [DTSS-620]

### PR DESCRIPTION
**Pull request by george-e-shaw-iv**
**What this PR does / why we need it**: 
* Allow for more granular trace names whilst also being able to
distinguish between trace types that are important to us.

**JIRA ID**: [DTSS-620]

**Notes for your reviewer**:

[DTSS-620]: https://outreach-io.atlassian.net/browse/DTSS-620

### CODE REVIEW

**Review by jaredallard — APPROVED** [review]
Seems like a solid approach.

File: pkg/trace/call.go [hunk 1]
```diff
@@ -150,23 +120,37 @@ func StartCall(ctx context.Context, cType callType, args ...log.Marshaler) conte
...
+func StartExternalCall(ctx context.Context, cType string, args ...log.Marshaler) context.Context {
 	ctx = StartCall(ctx, cType, args...)
 	ctx.Value(infoKey).(*callInfo).kind = metrics.CallKindExternal
 
 	return ctx
 }
 
+// SetTypeGRPC is meant to set the call type to GRPC on a context that has
+// already been initialized for tracing via StartCall or StartExternalCall.
+func SetTypeGRPC(ctx context.Context) context.Context {
```
Comment (vk-outreach) [inline pkg/trace/call.go:142]:
```suggestion
func SetCallTypeGRPC(ctx context.Context) context.Context {
```

File: pkg/trace/call.go [hunk 2]
```diff
@@ -150,23 +120,37 @@ func StartCall(ctx context.Context, cType callType, args ...log.Marshaler) conte
...
+	ctx.Value(infoKey).(*callInfo).callType = callTypeGRPC
+	return ctx
+}
+
+// SetTypeHTTP is meant to set the call type to HTTP on a context that has
+// already been initialized for tracing via StartCall or StartExternalCall.
+func SetTypeHTTP(ctx context.Context) context.Context {
```
Comment (vk-outreach) [inline pkg/trace/call.go:149]:
```suggestion
func SetCallTypeHTTP(ctx context.Context) context.Context {
```

**Review by vk-outreach — APPROVED** [review]
LGTM but I'd make the setters explicitly target `Calls`.

Also, it would be nice to just migrate all the call related things to its own package soon...

**Pull request merged**